src/generators.rs

```diff
@@ -181,6 +181,32 @@ impl BulletproofGens {
         }
     }
 
+    /// Precomputes additional generators and increases the capacity.
+    pub fn add_gens(&mut self, add_gens: usize) {
```

Comment:
bikeshed: let's spell out words fully and maybe name it more like `increase_capacity(new_capacity: ...)`? This way we don't need to compute how much extra items we need - we can just set exactly how much we need. No-op if capacity is sufficient.